lotman/lotman_ui.go

```diff
@@ -0,0 +1,573 @@
+	}
+	scopes := strings.Split(scope, " ")
+	for _, scope := range scopes {
+		if scope == token_scopes.Lot_Create.String() {
+			tokenApproved = true
+			break
+		}
+	}
+
+	if !tokenApproved {
+		return false, errors.New("The token was correctly signed but did not possess the necessary lot.create scope")
+	}
+
+	// At this point, we have a good token, now we need to get the appropriate owner for the lot.
+	// To do this, we get a namespace from the path and then get the issuer for that namespace. If no
+	// namespace exists, we'll assign ownership to the federation.
+
+	// TODO: Once we have the new Director endpoint that returns a namespace for a given path, we'll use that
+	// and cut out a lot of this cruft
+
+	// Get the namespace by querying the director and checking the headers
+	errMsgPrefix := "the provided token is acceptible, but no owner could be determined because "
+	directorUrlStr := param.Federation_DirectorUrl.GetString()
+	if directorUrlStr == "" {
+		return false, errors.New(errMsgPrefix + "the federation director URL is not set")
+	}
+	directorUrl, err := url.Parse(directorUrlStr)
+	if err != nil {
+		return false, errors.Wrap(err, errMsgPrefix+"the federation director URL is not a valid URL")
+	}
+
+	directorUrl.Path, err = url.JoinPath("/api/v1.0/director/object", path)
+	if err != nil {
+		return false, errors.Wrap(err, errMsgPrefix+"the director URL could not be joined with the path")
+	}
+
+	// Get the namespace by querying the director and checking the headers. The client should NOT
+	// follow the redirect
+	httpClient := &http.Client{
+		Transport: config.GetTransport(),
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	req, err := http.NewRequest("GET", directorUrl.String(), nil)
+	if err != nil {
+		return false, errors.Wrap(err, errMsgPrefix+"the director request could not be created")
+	}
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return false, errors.Wrapf(err, errMsgPrefix+"the director couldn't be queried for path %s", path)
+	}
+
+	// Check the response code, make sure it's not in the error ranges (400-500)
+	if resp.StatusCode < 200 || resp.StatusCode >= 400 {
+		return false, errors.Errorf(errMsgPrefix+"the director returned a bad status for path %s: %s", path, resp.Status)
+	}
+
+	// Get the namespace from the X-Pelican-Namespace header
+	namespaceHeader := resp.Header.Values("X-Pelican-Namespace")
+	xPelicanNamespaceMap := utils.HeaderParser(namespaceHeader[0])
+	namespace := xPelicanNamespaceMap["namespace"]
+
+	// Get the issuer URL for that namespace
+	nsIssuerUrl, err := server_utils.GetNSIssuerURL(namespace)
+	if err != nil {
+		return false, errors.Wrapf(err, errMsgPrefix+"no issuer could be found for namespace %s", namespace)
+	}
+	lot.Owner = nsIssuerUrl
+
+	return true, nil
+}
+
+func VerifyDeleteLotToken(lotName string, strToken string) (bool, error) {
```

Comment:
All my comments below apply to here. In fact, this is a lot of duplicated code. Really the only difference between this and VerifyUpdateDeleteLotToken is the first debug message and the scope. I would make this one function and just pass the scope you are trying to match rather than duplicate the whole thing.